torch/csrc/jit/ir/ir.cpp

```diff
@@ -1048,6 +1048,8 @@ bool Node::hasSideEffects() const {
     case prim::Guard:
     case prim::rpc_async: // It represents RPC message sent.
     case aten::wait: // It can represent RPC message received.
+    case prim::Enter:
```

Comment:
Can you add description of your PRs changes where applicable to  https://github.com/pytorch/pytorch/blob/f4532d7542cc819786af9a3feb747be60d3de2b9/torch/csrc/jit/OVERVIEW.md ?

and also edit https://github.com/pytorch/pytorch/blob/master/docs/source/jit_python_reference.rst to say we support with statements now

Reply:
Done. Although I would be more comfortable doing this after more testing or at least making sure `torch.no_grad` works.

Reply:
Yea, good point. up to you